include/odp/api/spec/traffic_mngr.h

```diff
@@ -1666,6 +1680,19 @@ int odp_tm_enq(odp_tm_queue_t tm_queue, odp_packet_t pkt);
  */
 int odp_tm_enq_with_cnt(odp_tm_queue_t tm_queue, odp_packet_t pkt);
 
+/** The odp_tm_queue_enq() function is used to add packets to a given TM system.
+ * This API needs to be used to enqueue events of type ODP_EVENT_PACKET_VECTOR
+ * or ODP_EVENT_PACKET to tm queue only when PKTOUT mode is
+ * ODP_PKTOUT_MODE_TM_QUEUE. Apart from this, rest of the behaviour is
+ * identical to that of odp_tm_enq().
+ *
+ * @param tm_queue  Specifies the tm_queue (and indirectly the TM system).
+ * @param ev        Event handle representing a packet or packet vector.
+ * @return          Returns 0 upon success, < 0 upon failure. One of the
+ *                  more common failure reasons is WRED drop.
+ */
+int odp_tm_queue_enq(odp_tm_queue_t tm_queue, odp_event_t ev);
```

Comment:
This seems to be unrelated to everything else in this PR. Adding vector support in TM output should be in a separate PR, but I do not even see clear need for it, especially after the enqueue_multi gets added. Vectors are now only used in packet input to basically tie a set of packets into single event and thus single scheduling context. I do not see similar need in packet output.

Reply:
odp_queue_enq() already accepts ODP_EVENT_PACKET_VECTOR type of events even in linux-generic(pktout_enqueue()). So, this should be same, behavior I believe ?

Reply:
As discussed, having a TM output function that takes events as input is unrelated to the rest of the changes. In particular, output with input-order-preservation does not require it. Let's not add it now. It will be possible to add it later in the API if needed. IPsec does not have such event-taking output function either.